File: app/schemas/observability.py
"""Tenant-scoped operational summary contracts."""

from datetime import datetime

from pydantic import BaseModel


class WorkflowCounts(BaseModel):
    workflow: str
    runs: int
    gate_passed: int
    failed: int


class DailyCounts(BaseModel):
    date: str
    runs: int
    case_errors: int


class EvaluationCounts(BaseModel):
    runs: int
    completed: int
    failed: int
    running: int
    gates_passed: int
    live_runs: int
    replay_runs: int
    results: int
    case_errors: int
    cost_usd_known: float | None
    cost_reported_results: int
    cost_unknown_results: int
    tokens_input_known: int | None
    tokens_output_known: int | None
    tokens_reported_results: int
    tokens_unknown_results: int
    input_tokens_reported_results: int
    input_tokens_unknown_results: int
    output_tokens_reported_results: int
    output_tokens_unknown_results: int
    avg_latency_ms_known: float | None
    latency_reported_results: int
    latency_unknown_results: int
    by_workflow: list[WorkflowCounts]
    daily: list[DailyCounts]


class IngestionCounts(BaseModel):
    runs: int
    completed: int
    failed: int
    partial: int
    partial_with_errors: int
    running: int
    records_seen: int
    records_failed: int
    stalled_runs: int


class AttentionItem(BaseModel):
    code: str
    level: str
    summary: str
    count: int
    href: str


class ObservabilityOverview(BaseModel):
    generated_at: datetime
    window_start: datetime
    window_end: datetime
    days: int
    evaluations: EvaluationCounts
    ingestion: IngestionCounts
    attention: list[AttentionItem]

File: src/types/observability.ts
export type ObservabilityDays = 1 | 7 | 30;

export interface ObservabilityOverview {
  generated_at: string;
  window_start: string;
  window_end: string;
  days: number;
  evaluations: {
    runs: number; completed: number; failed: number; running: number;
    gates_passed: number; live_runs: number; replay_runs: number;
    results: number; case_errors: number;
    cost_usd_known: number | null; cost_reported_results: number; cost_unknown_results: number;
    tokens_input_known: number | null; tokens_output_known: number | null;
    tokens_reported_results: number; tokens_unknown_results: number;
    input_tokens_reported_results: number; input_tokens_unknown_results: number;
    output_tokens_reported_results: number; output_tokens_unknown_results: number;
    avg_latency_ms_known: number | null; latency_reported_results: number; latency_unknown_results: number;
    by_workflow: { workflow: string; runs: number; gate_passed: number; failed: number }[];
    daily: { date: string; runs: number; case_errors: number }[];
  };
  ingestion: {
    runs: number; completed: number; partial: number; partial_with_errors: number; failed: number; running: number;
    records_seen: number; records_failed: number; stalled_runs: number;
  };
  attention: { code: string; level: string; summary: string; count: number; href: string }[];
}
